ksqldb-engine/src/main/java/io/confluent/ksql/internal/MetricsReporter.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.internal;
+
+import com.google.common.collect.ImmutableMap;
+import io.confluent.common.Configurable;
+
+import java.io.Closeable;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+/**
+ * This interface is used to report metrics as data points to a
+ * metrics framework.
+ * The implementations of this interface specify how to report to
+ * a specific metrics framework (e.g. JMX or Confluent's Telemetry
+ * pipeline)
+ */
+public interface MetricsReporter extends Closeable, Configurable {
+
+  /**
+   * A data point that should be reported.
+   */
+  class DataPoint {
+    private final String name;
+    private final Instant time;
```

Comment:
@cadonna sorry I didn't catch this earlier - but why did you pick `Instant` here? JW because right now the metrics reporter skeleton usese `Time` / `long`